### PR TITLE
backend/session/logind: improve logging

### PR DESCRIPTION
**Pull request by emersion**
Will maybe help with https://github.com/swaywm/sway/issues/3136

### CODE REVIEW

File: backend/session/logind.c [hunk 1]
```diff
@@ -94,27 +95,27 @@ static int logind_take_device(struct wlr_session *base, const char *path) {
 error:
 	sd_bus_error_free(&error);
 	sd_bus_message_unref(msg);
-	return fd;
+	return -1;
```
Comment (kragniz) [inline backend/session/logind.c]: why did you change this to always return -1?

File: backend/session/logind.c [hunk 2]
```diff
@@ -70,22 +69,24 @@ static int logind_take_device(struct wlr_session *base, const char *path) {
...
 		goto error;
 	}
 
+	int fd = -1;
 	int paused = 0;
 	ret = sd_bus_message_read(msg, "hb", &fd, &paused);
 	if (ret < 0) {
 		wlr_log(WLR_ERROR, "Failed to parse D-Bus response for '%s': %s",
-			path, strerror(-ret));
+			path, error.message);
```
Comment (progandy) [inline backend/session/logind.c]: ```error.message``` is only set by ```sd_bus_call_method```. Why would you use it here?

File: backend/session/logind.c (hunk 1 above)
Comment (progandy) [reply backend/session/logind.c]: @ddevault It is not a pure error branch, but a *finally* branch.

Comment (emersion) [reply backend/session/logind.c]: Oh right. I'll s/error/out/ because it's confusing.

File: backend/session/logind.c (hunk 2 above)
Comment (emersion) [reply backend/session/logind.c]: Good catch

File: backend/session/logind.c [hunk 3]
```diff
@@ -70,28 +69,30 @@ static int logind_take_device(struct wlr_session *base, const char *path) {
 		session->path, "org.freedesktop.login1.Session", "TakeDevice",
 		&error, &msg, "uu", major(st.st_rdev), minor(st.st_rdev));
 	if (ret < 0) {
-		wlr_log(WLR_ERROR, "Failed to take device '%s': %s", path, error.message);
-		goto error;
+		wlr_log(WLR_ERROR, "Failed to take device '%s': %s", path,
+			error.message);
+		goto out;
```
Comment (progandy) [inline backend/session/logind.c]: ```fd``` is not defined if this goto is triggered, but it is needed for the return.

Comment (emersion) [reply backend/session/logind.c]: Yeah, just pushed a fix

**Comment by ddevault** [comment]
Thanks!
